lib/bugsnag_http_client.dart

```diff
@@ -189,9 +266,18 @@ class Client extends http.BaseClient{
 
   @override
   Future<Uint8List> readBytes(Uri url, {Map<String, String>? headers}) async {
-    var requestId = _sendRequestStartNotification(url.toString(),"READ");
+    var requestId = _sendRequestStartNotification(url.toString(), "READ");
     try {
-      var response = await get(url, headers: headers);
+      var allHeaders = headers ?? {};
+      _headersProvider
+          .requestHeaders(
+        url: url.toString(),
+        requestId: requestId,
+      )
+          ?.forEach((key, value) {
+        allHeaders[key] = value;
+      });
```

Comment:
Could we not use the spread operator here:

```suggestion
     final allHeaders = {
          ...?headers,
          ...?(_headersProvider.requestHeaders(
                  url: url.toString(),
                  requestId: requestId,
              ),
          ),
      };
```